src/wazuh_modules/agent_upgrade/manager/wm_agent_upgrade_manager.h

```diff
@@ -0,0 +1,157 @@
+/*
+ * Wazuh Module for Agent Upgrading
+ * Copyright (C) 2015-2020, Wazuh Inc.
+ * July 30, 2020.
+ *
+ * This program is free software; you can redistribute it
+ * and/or modify it under the terms of the GNU General Public
+ * License (version 2) as published by the FSF - Free Software
+ * Foundation.
+ */
+
+#ifndef WM_AGENT_UPGRADE_MANAGER_H
+#define WM_AGENT_UPGRADE_MANAGER_H
+
+#define WM_UPGRADE_MINIMAL_VERSION_SUPPORT "v3.0.0"
+#define WM_UPGRADE_NEW_VERSION_REPOSITORY "v3.4.0"
+#define WM_UPGRADE_NEW_UPGRADE_MECHANISM "v4.1.0"
+#define WM_UPGRADE_WPK_DEFAULT_PATH "var/upgrade/"
+#define WM_UPGRADE_WPK_DOWNLOAD_TIMEOUT 60000
+#define WM_UPGRADE_WPK_DOWNLOAD_ATTEMPTS 5
+#define WM_UPGRADE_WPK_OPEN_ATTEMPTS 10
+#define MANAGER_ID 0
+
+typedef enum _wm_upgrade_error_code {
+    WM_UPGRADE_SUCCESS = 0,
+    WM_UPGRADE_PARSING_ERROR,
+    WM_UPGRADE_PARSING_REQUIRED_PARAMETER,
+    WM_UPGRADE_TASK_CONFIGURATIONS,
+    WM_UPGRADE_TASK_MANAGER_COMMUNICATION,
+    WM_UPGRADE_TASK_MANAGER_FAILURE,
+    WM_UPGRADE_GLOBAL_DB_FAILURE,
+    WM_UPGRADE_INVALID_ACTION_FOR_MANAGER,
+    WM_UPGRADE_AGENT_IS_NOT_ACTIVE,
+    WM_UPGRADE_UPGRADE_ALREADY_IN_PROGRESS,
+    WM_UPGRADE_NOT_MINIMAL_VERSION_SUPPORTED,
+    WM_UPGRADE_SYSTEM_NOT_SUPPORTED,
+    WM_UPGRADE_URL_NOT_FOUND,
+    WM_UPGRADE_WPK_VERSION_DOES_NOT_EXIST,
+    WM_UPGRADE_NEW_VERSION_LEES_OR_EQUAL_THAT_CURRENT,
+    WM_UPGRADE_NEW_VERSION_GREATER_MASTER,
+    WM_UPGRADE_WPK_FILE_DOES_NOT_EXIST,
+    WM_UPGRADE_WPK_SHA1_DOES_NOT_MATCH,
+    WM_UPGRADE_UNKNOWN_ERROR
+} wm_upgrade_error_code;
+
+typedef enum _wm_upgrade_command {
+    WM_UPGRADE_UPGRADE = WM_TASK_UPGRADE,
+    WM_UPGRADE_UPGRADE_CUSTOM = WM_TASK_UPGRADE_CUSTOM,
+    WM_UPGRADE_AGENT_GET_STATUS = WM_TASK_UPGRADE_GET_STATUS,
+    WM_UPGRADE_AGENT_UPDATE_STATUS = WM_TASK_UPGRADE_UPDATE_STATUS
+} wm_upgrade_command;
+
+/**
+ * Definition of upgrade task to be run
+ * */
```

Comment:
Multiline comments use to be closed with `*/` simply.

Reply:
I've changed this